じゃんけんプログラムの作成

takaoooooo opened the pull request:
# じゃんけんプログラムの作成
次の機能を実装しましたので、レビューをお願いします。

## RockPaperScissorsクラスを作成
- じゃんけんをするためのクラスを作成
## initializeメソッドの作成
- 押されたキーを受け取り、fightメソッドへ渡す
## fightメソッドを作成
- じゃんけんを実行し、結果を表示する
## 実行結果
```
ruby main.rb
何本勝負？(press 1 or 3 or 5)
3
3本勝負を選びました
じゃんけん…(press g or c or p)
g
CPU…パー
あなた…グー
負け!
0勝1敗
じゃんけん…(press g or c or p)
じゃんけんプログラムの作成
g
CPU…チョキ
あなた…グー
勝ち!
1勝1敗
じゃんけん…(press g or c or p)
g
CPU…グー
あなた…グー
あいこで...(press g or c or p)
g
CPU…パー
あなた…グー
負け!
1勝2敗
結果
1勝2敗であなたの負け
```

takaoooooo committed:
Merge remote-tracking branch 'origin/master'

takaoooooo committed:
じゃんけんプログラムの作成

DaichiSaito commented on rock_paper_scissors/main.rb:
```diff
@@ -0,0 +1,65 @@
+    
+    # 押されたキーを取得
+    # @keyをfightメソッドに渡す
+    @key = gets.chomp!.to_i
```
keyという変数名は少々わかりづらいかなと思いました。

また、`gets.~~`はRockPaperScissorsクラスの中ではなく外でやったほうが自然かも？
```rb
game_count = gets.chomp!.to_i
# 1 or 3 or 5以外は受け付けない処理の記述は省略
rock_paper_scissors = RockPaperScissors.new(game_count)
```
みたいな感じ。

takaoooooo replied:
次のとおり修正しましたので、レビューをお願いします。

- 変数keyをgame_countにわかりやすく名称変更
- geme_countの取得（gets.~）の処理をクラス外に移動
- 上の修正に伴い、クラスの初期化処理を削除

takaoooooo committed:
変数keyをgame_countに名称変更

変数keyをgame_countにわかりやすく名称変更
geme_countの取得（gets.~）の処理をクラス外に移動
上の修正に伴い、クラスの初期化処理を削除

takaoooooo committed:
オブジェクト志向っぽいコードを写経

miketa-webprgr commented on rock_paper_scissors/janken.rb:
```diff
@@ -0,0 +1,185 @@
+  def create_hand
+    puts 'じゃんけん…(press g or c or p)'
+    # 入力された値をシンボルに変換してtypeに格納
+    type = gets.chomp.to_sym
```
細かいところですけど、typeという命名だと分かりづらいかなと思いました。
全然自信ないですけど、`janken_input`であるとか、具体的なものにしてはどうでしょう？

miketa-webprgr replied:
ただ、他のところでも共通でHandクラスなんかでもtypeというattributeにしているんですね。
なるほど。じゃんけんの種類、なんていう命名にするか、むずいですね笑
